distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/ServiceManagerImpl.java

```diff
@@ -362,68 +287,55 @@ public void startBundle(String bundleSymbolicName) throws BundleException {
     waitForRequiredBundles("");
   }
 
   @Override
-  public void waitForRequiredBundles(String symbolicNamePrefix) throws InterruptedException {
-    boolean ready = false;
-    if (bundleService == null) {
-      bundleService = getService(BundleService.class);
-    }
+  public void waitForRequiredBundles(String symbolicNamePrefix) {
+    await("Waiting for bundles with prefix to be ready: " + symbolicNamePrefix)
+        .atMost(FEATURES_AND_BUNDLES_TIMEOUT, TimeUnit.MILLISECONDS)
+        .pollDelay(1, TimeUnit.SECONDS)
+        .until(
+            () ->
+                Arrays.stream(bundleContext.getBundles())
+                    .filter(bundle -> bundle.getSymbolicName().startsWith(symbolicNamePrefix))
+                    .allMatch(this::isBundleReady));
+  }
 
-    long timeoutLimit = System.currentTimeMillis() + FEATURES_AND_BUNDLES_TIMEOUT;
-    while (!ready) {
-      List<Bundle> bundles = Arrays.asList(getBundleContext().getBundles());
-
-      ready = true;
-      for (Bundle bundle : bundles) {
-        if (bundle.getSymbolicName().startsWith(symbolicNamePrefix)) {
-          String bundleName = bundle.getHeaders().get(Constants.BUNDLE_NAME);
-          BundleInfo bundleInfo = bundleService.getInfo(bundle);
-          BundleState bundleState = bundleInfo.getState();
-          if (bundleInfo.isFragment()) {
-            if (!BundleState.Resolved.equals(bundleState)) {
-              LOGGER.info("{} bundle not ready yet", bundleName);
-              ready = false;
-            }
-          } else if (bundleState != null) {
-            if (BundleState.Failure.equals(bundleState)) {
-              printInactiveBundles();
-              fail("The bundle " + bundleName + " failed.");
-            } else if (!BundleState.Active.equals(bundleState)) {
-              LOGGER.info("{} bundle not ready with state {}", bundleName, bundleState);
-              ready = false;
-            }
-          }
-        }
-      }
+  private boolean isBundleReady(Bundle bundle) {
+    String name = bundle.getHeaders().get(Constants.BUNDLE_NAME);
+    BundleInfo info = bundleService.getInfo(bundle);
+    BundleState state = info.getState();
 
-      if (!ready) {
-        if (System.currentTimeMillis() > timeoutLimit) {
-          printInactiveBundles();
-          fail(
-              String.format(
-                  "Bundles and blueprint did not start within %d minutes.",
-                  TimeUnit.MILLISECONDS.toMinutes(FEATURES_AND_BUNDLES_TIMEOUT)));
-        }
-        LOGGER.info("Bundles not up, sleeping...");
-        Thread.sleep(1000);
+    boolean ready;
+    if (info.isFragment()) {
+      ready = BundleState.Resolved.equals(state);
+    } else {
+      if (BundleState.Failure.equals(state)) {
+        printInactiveBundles();
+        LOGGER.error("The bundle " + name + " failed.");
       }
+      ready = BundleState.Active.equals(state);
+    }
+
+    if (!ready) {
+      LOGGER.info("{} bundle not ready yet", name);
```

Comment:
Since this is called in a wait loop and might flood the logs a little, do we want to to put it a little lower to debug?